TeleSharp.Generator/Program.cs

```diff
@@ -200,11 +376,36 @@ static void Main(string[] args)
                     classFile.Close();
                 }
             }
+            Console.WriteLine("Adding fields to abstract classes...");
+            // add fields to abstract classes
+            foreach (KeyValuePair<string, List<Property>> absClass in abstractParams)
+            {
+                if(absClass.Key == "himself")
+                    throw new InvalidOperationException("ARGH! It was a class without a parent, why it came into the list? :|");
+                string path = Path.Combine(outputPath, GetNameSpace(absClass.Key, TargetNamespace).Replace(TargetNamespace, @"TL\").Replace(".", ""), GetNameofClass(absClass.Key, true) + ".cs").Replace(@"\\", @"\");
+                string tmp = File.ReadAllText(path);
+                tmp = tmp.Replace("/* PARAMS */", ConvertPropertyList(absClass.Value));
+                File.WriteAllText(path, tmp);
+            }
+
+            #endregion
+            Console.WriteLine("Done.");
+        }
+
+        public static string ConvertPropertyList(List<Property> list)
+        {
+            string output = "";
+            foreach (var property in list)
+            {
+                output += $"        public {property.type} {property.name} {{ get; set; }}" + Environment.NewLine;
+            }
+            return output;
         }
+
         public static string FormatName(string input)
         {
             if (String.IsNullOrEmpty(input))
-                throw new ArgumentException("ARGH!");
+                throw new ArgumentException("ARGH! Class Name was empty.");
```

Comment:
I thought the exception message didn't explain anything, so I decided to add some words in there to clarify the throwing reason.